crates/trie/src/metrics.rs

```diff
@@ -0,0 +1,69 @@
+use metrics::Histogram;
+use reth_metrics::Metrics;
+use std::time::Instant;
+
+#[derive(Metrics)]
+#[metrics(scope = "trie")]
+pub(crate) struct TrieRootMetrics {
+    /// The number of microseconds trie root calculation lasted.
+    duration_micros: Histogram,
+    /// The number of branches added during trie root calculation.
+    branches_added: Histogram,
+    /// The number of leaves added during trie root calculation.
+    leaves_added: Histogram,
+}
+
+impl TrieRootMetrics {
+    pub(crate) fn new(ty: TrieType) -> Self {
+        Self::new_with_labels(&[("type", ty.as_str())])
+    }
+
+    pub(crate) fn record(&self, tracker: TrieTracker) {
+        self.duration_micros.record(tracker.started_at.elapsed().as_millis() as f64);
```

Comment:
millis or micros? Also noticing that you can record a duration which gets saved as secs_f64: <https://docs.rs/metrics/latest/src/metrics/common.rs.html#253-257>

Reply:
agreed, we should record it as floating-point seconds as we do throughout the codebase. It has the precision up to nanos.